### PR TITLE
Improved svds documentation

### LINKED ISSUE #305: `svds` documented return type is wrong
The Julia manual describes `svds` with this signature:

```
svds(A; nsv=6, ritzvec=true, tol=0.0, maxiter=1000) -> (left_sv, s, right_sv, nconv, niter, nmult, resid)
```

If you look into the actual code in `arnoldi.jl`, you see that the return type depends on the value of the argument `ritzvec`; if false, there are only 5 instead of 7 return values.

### LINKED ISSUE #14838: `svds` documented return type is wrong
The Julia manual describes `svds` with this signature:
```
svds(A; nsv=6, ritzvec=true, tol=0.0, maxiter=1000) -> (left_sv, s, right_sv, nconv, niter, nmult, resid)
```
If you look into the actual code in `arnoldi.jl`, you see that the return type depends on the value of the argument `ritzvec`; if false, there are only 5 instead of 7 return values.

### PR DESCRIPTION
**Pull request by jiahao**
- Moves `svds` docstring from `docs/helpdb/Base.jl` to
  `base/linalg/arnoldi.jl`.
- Corrects description of `svds` - it uses implicitly restart Lanczos
  only, never the implicitly restarted Arnoldi method.
- Describes each component of the output tuple, emphasizing that the
  singular vectors are not returned if `ritzvec=false`.
- Explains the formal relation between `svds` and `eigs`

Closes JuliaLang/LinearAlgebra.jl#305

### CODE REVIEW

File: base/linalg/arnoldi.jl [hunk 1]
```diff
@@ -278,6 +278,48 @@ function svds{T}(A::AbstractMatrix{T}; kwargs...)
...
+
+"""
+    svds(A; nsv=6, ritzvec=true, tol=0.0, maxiter=1000) -> ([left_sv,] s, [right_sv,] nconv, niter, nmult, resid)
+
+Computes the largest singular values `s` of `A` using implicitly restart Lanczos
+iterations derived from [`eigs`](:func:`eigs`).
+
+Inputs:
+
+* `A`: Linear operator. It can either subtype of `AbstractArray` (e.g., sparse matrix) or
```
Comment (tkelman) [inline base/linalg/arnoldi.jl]: be a subtype?

File: base/linalg/arnoldi.jl [hunk 2]
```diff
@@ -278,6 +278,48 @@ function svds{T}(A::AbstractMatrix{T}; kwargs...)
...
+* `nsv`: Number of singular values.
+* `ritzvec`: If `true`, return the left and right singular vectors `left_sv` and `right_sv`.
+   If `false`, omit the singular vectors. Default: `true`.
+* `tol`: tolerance, see [`eigs`](:func:`eigs`).
+* `maxiter`: Maximum number of iterations, see [`eigs`](:func:`eigs`).
+
+Outputs:
+
+* `left_sv`: Left singular vectors (only if `ritzvec = true`).
+* `s`: A vector of length `nsv` containing the requested singular values,
```
Comment (tkelman) [inline base/linalg/arnoldi.jl]: this line ends with comma, the others period?

File: base/linalg/arnoldi.jl (hunk 1 above)
Comment (jiahao) [inline base/linalg/arnoldi.jl]: Ah yes. I didn't read the original carefully enough

File: base/linalg/arnoldi.jl [hunk 3]
```diff
@@ -278,6 +278,48 @@ function svds{T}(A::AbstractMatrix{T}; kwargs...)
...
+* `right_sv`: Right singular vectors (only if `ritzvec = true`).
+* `nconv`: number of converged singular values.
```
Comment (tkelman) [inline base/linalg/arnoldi.jl]: upper case Number for consistency with other bullets here?

**Pull request merged**